### DISCUSSION TITLE
Private GitHub Pages redirects to internal url

### DISCUSSION THREAD

**Discussion by JeroenKnoops** [community/community#21581] [Other Feature Feedback, Questions, & Ideas]
<p>We want to use Private GitHub Pages, but we found out the nice url: ( https://.github.io/ ) redirects to an internal url, which is not really what we want.</p>
<p>We also cannot seem to get the custom domain working for internal pages.</p>
<p>Is it possible to NOT redirect to the internal url?</p>
<p>Example urls:<br>
nice url: <a href="https://philips-internal.github.io/orbit" rel="noopener nofollow ugc">https://philips-internal.github.io/orbit</a><br>
redirected to url: <a href="https://congenial-funicular-829fc4f5.pages.github.io/" rel="noopener nofollow ugc">https://congenial-funicular-829fc4f5.pages.github.io/</a></p>

**Comment by otter-computer** [comment, marked as answer]
<p>It’s not possible to change this, no. The <code>pages.github.io</code> URL needs to be used so GitHub can host an authentication layer in front of your site and ensure that the subsequent site is secure. For more information I’d recommend reading <a href="https://github.community/t/disable-forwarding-from-orga-github-io-repo-to-foo-bar-pages-github-io/161815/2">this comment (and subsequent reply) from one of our engineers</a>.</p>
<p>The only current solution for prettifying your internal URL is to <a href="https://docs.github.com/en/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site">use a custom domain</a>.</p>

**Comment by JeroenKnoops** [comment]
<p>The custom domain is working, but it has a <strong>serious security bug</strong>…</p>
<p>Let’s pretend I created a custom domain called: “<a href="http://my-custom-domain.com" rel="noopener nofollow ugc">my-custom-domain.com</a>”<br>
I’ve created a CNAME so all sub-domains are forwarded to my organizations <a href="http://github.io" rel="noopener nofollow ugc">github.io</a> page.</p>
<p>*.my-custom-domain.com → <a href="http://philips-internal.github.io" rel="noopener nofollow ugc">philips-internal.github.io</a></p>
<p>Now people can add a custom domain name with sub-domain (like <a href="http://documentation.my-custom-domain.com" rel="noopener nofollow ugc">documentation.my-custom-domain.com</a>) to their internal github pages, but… If they do something wrong (like forgetting to code the CNAME in the deploy directory), the custom domain name is not set anymore. Now the url will return in a 404… Also when archiving the repository, the url results in a 404.</p>
<p>What we saw today, is other people <strong>outside</strong> of our organization claiming the sub domain and there is no way of reverting that. The attacker just took over the url. In our case today, he put a message to contact him, so we’ve fixed it all, but I think one thing can be improved to the way GitHub has implemented this.</p>
<p>When a subdomain is used, and it is aliased to ‘<a href="http://philips-internal.github.io" rel="noopener nofollow ugc">philips-internal.github.io</a>’ , github should only look into the repositories of organization “philips-internal” to resolve the repository by looking for a similar custom domain. Now it also looks for repositories outside of the organization, and this can be used to put information on an internal github page…</p>
<p>If you want to know more about this, please contact me…</p>
